Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,307 @@
+
+        /// <summary>
+        /// Searches all profiles and sub profiles for matching fields. Returns results in unranked order.
+        /// </summary>
+        /// <param name="profile">The configuration profile you're searching.</param>
+        /// <param name="config">Configuration for search.</param>
+        public static IEnumerable<ProfileSearchResult> SearchProfileFields(UnityEngine.Object profile, SearchConfig config)
+        {
+            if (string.IsNullOrEmpty(config.SearchFieldString) && config.SelectedSubjects == 0)
+            {   // If the config is empty, bail early
+                yield break;
+            }
+
+            // Generate keywords if we haven't yet
+            if (config.Keywords == null)
+            {
+                config.Keywords = new HashSet<string>(config.SearchFieldString.Split(new string[] { " ", "," }, StringSplitOptions.RemoveEmptyEntries));
+                config.Keywords.RemoveWhere(s => s.Length < minSearchStringLength);
+            }
+
+            // The result that we will return, if not empty
+            ProfileSearchResult result = new ProfileSearchResult();
+
+            // Ignore these requirements if criteria is empty
+            config.RequireAllKeywords &= config.Keywords.Count > 0;
+            config.RequireAllSubjects &= config.SelectedSubjects != 0;
+            // Subject requirements are met by the profile's subject tag
+            // If not all subjects are required, a profile with the wrong tags or no tags can still match
+            int profileMatchStrength = 0;
+            bool subjectReqsMet = CheckProfileSubjects(config, profile, out profileMatchStrength);            
+            if (subjectReqsMet)
+            { 
+                result.Profile = profile;
+                result.ProfileMatchStrength = profileMatchStrength;
+                result.Fields = new List<FieldSearchResult>();
+            }
+
+            // Otherwise go through the profile's serialized fields
+            SerializedProperty iterator = new SerializedObject(profile).GetIterator();
+            bool hasNextProperty = iterator.Next(true);
+
+            while (hasNextProperty)
+            {
+                if (serializedPropertiesToIgnore.Contains(iterator.name))
+                {   // We don't care about this property
+                    hasNextProperty = iterator.Next(false);
+                    continue;
+                }
+
+                // Check to see if this is a profile - if it is, enter the profile and check its children
+                bool isProfileField = false;
+                if (iterator.propertyType == SerializedPropertyType.ObjectReference && iterator.objectReferenceValue != null)
+                {   // If the property is itself a profile, perform a recursive search
+                    Type referenceType = iterator.objectReferenceValue.GetType();
+                    if (typeof(BaseMixedRealityProfile).IsAssignableFrom(referenceType))
```

Comment:
Good place for Utility function => IsProfile or IsConcreteProfile and then remove comment